File: Python/lengthOfLongestSubstring.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        l = 0; r = 0; res= 0
        Set = set()
        while r < len(s):
            if s[r] not in Set:
                Set.add(s[r])
                r += 1
                res = max(res, len(Set))
            else:
                Set.remove(s[l])
                l += 1

        return res

File: Java/lengthOfLongestSubstring.java
package Java;

import java.util.HashSet;
import java.util.Set;

public class lengthOfLongestSubstring {
    public int solution(String s) {
        int longest = 0, left = 0, right = 0;
        Set<Character> set = new HashSet<>();
        while (right < s.length()) {
            if (!set.contains(s.charAt(right))) {
                set.add(s.charAt(right));
                right++;
                longest = Math.max(longest, set.size());
            } else {
                set.remove(s.charAt(left));
                left++;
            }
        }
        return longest;
    }
}
